app/src/main/java/com/xmartlabs/scasas/doapp/ui/grouplist/GroupsListFragment.java

```diff
@@ -0,0 +1,474 @@
+  void onClikedTravelGroup() {
+    Group group = getGroup(R.string.travel);
+    goToListOfTasksFragment(group);
+  }
+
+  @OnClick(R.id.bills_linear)
+  void onClikedBillsGroup() {
+    Group group = getGroup(R.string.bills);
+    goToListOfTasksFragment(group);
+  }
+
+  @OnClick(R.id.auto_linear)
+  void onClikedAutoGroup() {
+    Group group = getGroup(R.string.auto);
+    goToListOfTasksFragment(group);
+  }
+
+  @OnClick(R.id.add_task)
+  void onClickedAddTextView() {
+    Group group = getGroup(groupsSpinnerView.getSelectedItem().toString());
+    Task task = Task.builder()
+        .title(titleView.getText().toString().trim())
+        .description(descriptionView.getText().toString().trim())
+        .date(LocalDate.now())
+        .isFinished(false)
+        .group(group)
+        .user(user)
+        .build();
+    insertTask(task, group);
+    hideSoftKeyboard();
+    setupCurrentDate();
+    getFinishedPercentage();
+    setupDateFields();
+  }
+
+  @OnClick(R.id.next_month)
+  void onClickedNextMonthButton() {
+    date = date.plusMonths(1);
+    getFinishedPercentage();
+    setupDateFields();
+  }
+
+  @OnClick(R.id.previous_month)
+  void onClickedPrevMonthButton() {
+    date = date.minusMonths(1);
+    getFinishedPercentage();
+    setupDateFields();
+  }
+
+  @OnClick(R.id.logout)
+  void onClickedLogoutButton() {
+    Intent intent = Henson.with(getContext())
+        .gotoSignInActivity()
+        .build()
+        .setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+    getContext().startActivity(intent);
+  }
+
+  private void setFieldsEmpty() {
+    titleView.setText(null);
+    descriptionView.setText(null);
+  }
+
+  private void closeNewTaskView() {
+    fabButtonView.setVisibility(View.VISIBLE);
+    newTaskView.setVisibility(View.GONE);
+    setFieldsEmpty();
+  }
+
+  private void adjustSpinnerView(Spinner spinner) {
+    TextView textView = (TextView) spinner.getSelectedView();
+    //noinspection deprecation
+    textView.setTextColor(getResources().getColor(R.color.white));
+    textView.setTextSize(2, 14);
+  }
+
+  private void goToListOfTasksFragment(Group group) {
+    Intent intent = Henson.with(getContext())
+        .gotoTasksListActivity()
+        .group(group)
+        .user(user)
+        .build();
+    getContext().startActivity(intent);
+  }
+
+  private void getGroups() {
+    groupController.getGroups(user).subscribe(new SingleSubscriber<List<Group>>() {
+      @Override
+      public void onSuccess(List<Group> groupList) {
+        groups = groupList;
+        updateAll(groupList);
+      }
+
+      @Override
+      public void onError(Throwable error) {
+        Timber.e(error);
+      }
+    });
+  }
+
+  private Group getGroup(@NonNull String groupName) {
+    return Stream.of(groups)
+        .filter(group -> Objects.equals(groupName, group.getName()))
+        .findFirst()
+        .get();
+  }
+
+  private Group getGroup(@StringRes int groupName) {
```

Comment:
I'd prefer to assign a numeric PK and use it, or if the groups are predefined use an enum